prow/gerrit/adapter/adapter.go

```diff
@@ -244,8 +244,17 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 			if shouldRun, err := postsubmit.ShouldRun(change.Branch, changedFiles); err != nil {
 				return fmt.Errorf("failed to determine if postsubmit %q should run: %v", postsubmit.Name, err)
 			} else if shouldRun {
+				postsubmitSpec := pjutil.PostsubmitSpec(postsubmit, refs)
+				// For postsubmit jobs, cloneref ignores `Refs.Pulls` because this is empty for github,
+				// for gerrit based repos, the commit SHA of the change is the same commit SHA merged into
+				// base branch, so using this for postsubmit jobs
```

Comment:
In particular it looks like the merge commit becomes a new patchset on the change when the change is submitted: https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#create-merge-patch-set-for-change

Reply:
yeah, added the link in comment